### PR TITLE
Update dhclient to non-blocking mode

### PR DESCRIPTION
**Pull request by lwindg**

### CODE REVIEW

File: ip/addr.py [hunk 1]
```diff
@@ -127,6 +117,28 @@ def ifupdown(iface, up):
...
+            sh.kill(dhclient)
+    except Exception as e:
+        _logger.info("Failed to stop dhclient: %s" % e)
+        pass
+
+    if enable:
+        if script:
+            sh.dhclient("-nw", "-sf", script, iface, _bg=True)
+        else:
+            sh.dhclient("-nw", iface, _bg=True)
```
Comment (imZack) [inline ip/addr.py]: should we have a timeout here?

Comment (lwindg) [inline ip/addr.py]: -nw cause dhclient become a daemon process (nowait). timeout is not required.

Comment (imZack) [inline ip/addr.py]: Daemon should not run with sh, use a service script?

Comment (lwindg) [inline ip/addr.py]: no service for `dhclient`, did you mean to create a script for `dhclient` instead of using `sh`?

Comment (imZack) [inline ip/addr.py]: yeah, unless dhclient will do technique like `fork, fork` change its parent pid to 1.

Comment (lwindg) [inline ip/addr.py]: I don't want `dhclient` to be separated from bundle; killed after bundle be killed is allowed.

**Pull request merged**